tidb-cloud/vector-search-integrate-with-langchain.md

```diff
@@ -56,61 +53,60 @@ from langchain_openai import OpenAIEmbeddings
 from langchain_text_splitters import CharacterTextSplitter
 ```
 
-### Step 3. Set up environments
+### Step 3. Set up your environment
 
-#### Step 3.1 Obtains the connection string to the TiDB cluster
-
-<SimpleTab>
-<div label="TiDB Serverless">
+#### Step 3.1 Obtain the connection string to the TiDB cluster
 
 1. Navigate to the [**Clusters**](https://tidbcloud.com/console/clusters) page, and then click the name of your target cluster to go to its overview page.
 
 2. Click **Connect** in the upper-right corner. A connection dialog is displayed.
 
 3. Ensure the configurations in the connection dialog match your operating environment.
 
-    - **Endpoint Type** is set to `Public`
-    - **Branch** is set to `main`
-    - **Connect With** is set to `SQLAlchemy`
+    - **Endpoint Type** is set to `Public`.
+    - **Branch** is set to `main`.
+    - **Connect With** is set to `SQLAlchemy`.
     - **Operating System** matches your environment.
 
-4. Switch to the **PyMySQL** tab and click the **Copy** icon to copy the connection string.
+4. Click the **PyMySQL** tab and copy the connection string.
 
-</div>
+    > **Tip:**
+    > 
+    > If you have not set a password yet, click **Generate Password** to generate a random password.
 
-</SimpleTab>
+#### Step 3.2 Configure environment variables
 
-#### Step 3.2 Configure the environment variables
+To establish a secure and efficient database connection, use the standard connection method provided by TiDB Cloud.
 
-Configure both the OpenAI and TiDB connection string settings that you will need.
+This document uses [OpenAI](https://platform.openai.com/docs/introduction) as the embedding model provider. In this step, you need to provide the connection string obtained from step 3.1 and your [OpenAI API key](https://platform.openai.com/docs/quickstart/step-2-set-up-your-api-key).
 
-In this notebook, we will follow the standard connection method provided by TiDB Cloud to establish a secure and efficient database connection.
+To configure the environment variables, run the following code. You will be prompted to enter your connection string and OpenAI API key:
 
 ```python
-# Here we useimport getpass
+# Use getpass to securely prompt for environment variables in your terminal.
 import getpass
 import os
 
-os.environ["OPENAI_API_KEY"] = getpass.getpass("OpenAI API Key:")
-# Copy from TiDB Cloud Console
-# The format of connection string: "mysql+pymysql://<USER>:<PASSWORD>@<HOST>:4000/<DB>?ssl_ca=/etc/ssl/cert.pem&ssl_verify_cert=true&ssl_verify_identity=true"
+# Copy your connection string from the TiDB Cloud console.
+# Connection string format: "mysql+pymysql://<USER>:<PASSWORD>@<HOST>:4000/<DB>?ssl_ca=/etc/ssl/cert.pem&ssl_verify_cert=true&ssl_verify_identity=true"
 tidb_connection_string = getpass.getpass("TiDB Connection String:")
+os.environ["OPENAI_API_KEY"] = getpass.getpass("OpenAI API Key:")
```

Comment:
Move the line `tidb_connection_string = getpass.getpass("TiDB Connection String:")` to a position before `getpass.getpass("OpenAI API Key:")`. This change aligns with the instructions in step 3.1 where the user has already copied the connection string.